napalm/ios/ios.py

```diff
@@ -581,14 +593,16 @@ def rollback(self):
         if not self._check_file_exists(cfg_file):
             raise ReplaceConfigException("Rollback config file does not exist")
         cmd = "configure replace {} force".format(cfg_file)
-        self._commit_handler(cmd)
+        output = self._commit_handler(cmd)
 
         # After a rollback - we no longer know whether this is configured or not.
         self.prompt_quiet_configured = None
 
         # Save config to startup
         self.device.save_config()
 
+        return output
+
```

Comment:
Why was this changed to return `output` from _commit_handler? That didn't seem necessary for the PR?

Reply:
@network-shark

Reply:
Bump @network-shark

Reply:
Okay, this was my misunderstanding...I don't think we can change this without changing the behavior of all the drivers, however (i.e. we would be changing what the `rollback` method returns).

Probably an easier solution is just to key off whether `rollback()` generates an exception or not (i.e. if `rollback` does not generate an exception then rollback worked...anything else should be a bug.

Reply:
@network-shark did you want to reply to Kirk's suggestion above?

Reply:
```
if "Invalid input detected" in output:
              err_header = "Configuration merge failed; automatic rollback attempted"
              merge_error = "{0}:\n{1}".format(err_header, output)

              try:
                  self.rollback()
              except:
                  raise
              else:
                  raise MergeConfigException(merge_error)
``` 
Like this ? @mirceaulinic  @ktbyers

Reply:
Probably something more like:

```python
    def rollback(self):
        """Rollback configuration to filename or to self.rollback_cfg file."""
        filename = self.rollback_cfg
        cfg_file = self._gen_full_path(filename)
        if not self._check_file_exists(cfg_file):
            raise ReplaceConfigException("Rollback config file does not exist")
        cmd = "configure replace {} force".format(cfg_file)
        self._commit_handler(cmd)
+      output = self._commit_handler(cmd)

        # After a rollback - we no longer know whether this is configured or not.
        self.prompt_quiet_configured = None

+      if "Rollback Done" not in output:
+          raise SomeException(f"Automatic rollback to previous configuration failed:\n\n{output}")

        # Save config to startup
        self.device.save_config()
```